etcd-client/src/kv/mod.rs

```diff
@@ -58,18 +65,21 @@ pub struct Kv {
     cache_enable: bool,
     /// Etcd watch request sender.
     watch_sender: Sender<EtcdWatchRequest>,
-    /// Flag to indicate if the cache is error and disabled
-    cache_err: AtomicBool,
+    /// Read write lock to indicate the availability of the cache.
+    /// When the cache has error, write lock is acquired to disable the cache and clean the cache.
+    /// Other operations will get read lock.
+    cache_availability: RwLock<()>,
```

Comment:
Another question is can we just use `cache_enable` here? When clean the cache, just set `cache_availability` to false, then after clean, set it to true

Reply:
It is possible that taskA read `cache_enable=true` and execute something and switch to taskB, taskB finds cache error and set `cache_enable=false` and clean cache. And then switch back to taskA to continue, taskA will either read empty or read some intermediate status.